herddb-core/src/main/java/herddb/core/TableSpaceManager.java

```diff
@@ -2022,6 +2027,28 @@ public TableChecksum createAndWriteTableCheksum(TableSpaceManager tableSpaceMana
         }
     }
 
+    public boolean checkpointNoWait(boolean full) {
+        if (checkpointInProgress.compareAndSet(false, true)
+                && System.currentTimeMillis() - lastTimeFinished.get() >= MIN_DURATION_BETWEEN_CHECKPOINTS) {
```

Comment:
I consider this very strange...
In anyone asked a checkpoint nowait that ended in 2 minutes and then another ask a new checkpoint nowait the second one is skipped (even if there isn't another checkpoint running).
But It will not be skipped if the first one didn't ask for a nowait (no nowait option... normal checkpoint...). So behaviour of this run is discated by previous requests parameters. It smells a little :/

I'd remove the MIN_DURATION_BETWEEN_CHECKPOINTS at all